tiledb/sm/misc/constants.h

```diff
@@ -432,6 +432,12 @@ extern const std::string vfsmode_write_str;
 /** The string representation for VFSMode append. */
 extern const std::string vfsmode_append_str;
 
+/** The maximum number of bytes to read-ahead in the VFS. */
```

Comment:
Should we make both of these config options? Seems like things that we might want to tweak when testing performance tuning.

Reply:
We can, if we're not too worried about the public configuration becoming too complex. The defaults are fairly liberal and will provide diminishing returns if increased. The default 10MB size seems negligible enough that a user wouldn't want to disable it.

What do you think?

Reply:
My line of thought was more if we are debugging or working with a user/customer then having this as an option is easier if we want to try a tweak then getting someone to recompile. I don't foresee this being touch much at all, and most users will leave the defaults. It was more about just allowing us the option to override in the future.

I think as we go the config options will continue to grow and be verbose but we can probably safely split the options into two categories, "Common things to change" and "Advanced", and split the config docs as such.

Reply:
Yeah, that's exactly what I was going to suggest: a "common" vs "advanced" parameters. That's easy enough to separate in public docs, but possibly we should structure them different in the code to make it more obvious when reading the available config options. We can save that for another day, I'll make these parameters configurable.